scalanet/src/io/iohk/scalanet/peergroup/PeerGroup.scala

```diff
@@ -48,14 +45,6 @@ trait Channel[A, M] {
     * The inbound stream of messages coming from the remote peer.
     */
   def in: ConnectableObservable[ChannelEvent[M]]
-
-  /**
-    * Terminate the Channel and clean up any resources.
-    *
-    * @return Implementations should wait for termination of the underlying network resources
-    *         and return a successful Task.
-    */
-  def close(): Task[Unit]
```

Comment:
I know it's a bit strange, but let's see how it works, we can always add it back. I found it more natural to do transformations and attach extra book-keeping that needs to happen when the channel is returned from the peer group. And it's much easier not to forget that to you have to call it because you don't have to remember it at all, or you have an extra variable that you have to think about how to use. Another benefit is that unless you go the advanced `.allocated` way there's just no way you'd call `sendMessage` on a closed channel, so overall finding out where a channel is closed becomes much easier.

Reply:
The thing is that restricting access to such fundamental operation on the upstream level, can lead to convoluted logic downstream just to get around those restrictions and just get the job done.  I know that, I would be highly unhappy if library forced me to go with this `.allocated` dance just to close channel. 

If you insist on it lets leave it as it is, but imo it just  passing the problem into the future i.e there will quickly be a task to add it back.

Reply:
Thanks, I hear you. I think it all depends on the use case, the way the downstream project is treating the connection as something that changes hands all the time and eventually gets closed at an unknown location doesn't fit this pattern very much. But I thought this library's headline says it's "resource managed", so it should take the conservative approach.